src/main/scala/com/evolutiongaming/prometheus/ObserveDuration.scala

```diff
@@ -1,65 +1,160 @@
 package com.evolutiongaming.prometheus
 
-import io.prometheus.client.Collector
+import io.prometheus.client.{Collector, SimpleTimer}
 
+import scala.annotation.{nowarn, unused}
 import scala.concurrent.{ExecutionContext, Future}
-
-trait ObserveDuration[F] {
-
+import scala.math.Numeric.Implicits.*
+import scala.util.Try
+
+/** Time duration measurement syntax for [[HasObserve]]-kind metrics from the prometheus client, i.e., Summary, Histogram.
+  *
+  * Here time is always reported in seconds, which means your prometheus metric name should end in `_seconds`.
+  *
+  * The class is not supposed to be used directly but as an implicit syntax provided by [[PrometheusHelper]].
+  */
+sealed trait ObserveDuration[F] {
+
+  /** Measures evaluation time of a block in seconds with nano-time precision
+    */
   def timeFunc[T](f: => T): T
-  def timeFuncNanos[T](f: => T): T
 
+  @deprecated(
+    message = "use timeFunc instead - it has nano-time precision now",
+    since = "1.1.0"
+  )
+  def timeFuncNanos[T](f: => T): T = timeFunc(f)
+
+  /** Measures evaluation time of an asynchronous block in seconds with nano-time precision
+    */
   def timeFuture[T](f: => Future[T]): Future[T]
-  def timeFutureNanos[T](f: => Future[T]): Future[T]
 
+  @deprecated(
+    message = "use timeFuture instead - it has nano-time precision now",
+    since = "1.1.0"
+  )
+  def timeFutureNanos[T](f: => Future[T]): Future[T] = timeFuture(f)
+
+  @deprecated(
+    message = "use timeTillNowMillis(: Long) with a primitive arg type and explicit precision name suffix",
+    since = "1.1.0"
+  )
   def timeTillNow[T](start: T)(implicit numeric: Numeric[T]): Unit
+
+  @deprecated(
+    message = "use timeTillNowNanos(: Long) with a primitive arg type",
+    since = "1.1.0"
+  )
   def timeTillNowNanos[T](start: T)(implicit numeric: Numeric[T]): Unit
-}
 
-object ObserveDuration {
-  def apply[F](implicit observeDuration: ObserveDuration[F]): ObserveDuration[F] = observeDuration
+  /** Measures in seconds the time spent since the provided start time obtained using [[ClockPlatform.nowMillis]]
+    */
+  def timeTillNowMillis(@unused startMs: Long): Unit = {
+    // default impl for a new method of a trait - added for keeping binary compatibility
+    // TODO: bincompat leftover, remove in 2.x
 
-  def fromHasObserver[F](observer: F)(implicit hasObserve: HasObserve[F], clock: ClockPlatform, ec: ExecutionContext): ObserveDuration[F] =
-    new ObserveDuration[F] {
+    timeTillNow[Long](startMs): @nowarn
+  }
 
-      override def timeFunc[T](f: => T): T =
-        measureFunction(f, clock.nowMillis, timeTillNow[Long])
+  /** Measures in seconds the time spent since the provided start time obtained using [[ClockPlatform.nowNano]]
+    */
+  def timeTillNowNanos(@unused startNs: Long): Unit = {
```

Comment:
Is it unused?

Reply:
Good catch! I missed it when I refactored the implementation, first it was a no-op.